File: Chapter 8/Chapter8ApplicationComplete/ViewController/public_html/resources/js/emailValidator.js
function EmailValidator(field) {
    this.firstNameField = field;
}
EmailValidator.prototype = new TrValidator();
EmailValidator.prototype.validate = function (value, label, converter) {
    var compareAgainst = AdfPage.PAGE.findComponent(this.firstNameField);
    if (value.indexOf(compareAgainst.getValue().toUpperCase().charAt(0)) != 0) {
        var fm = new TrFacesMessage("Validation Error", "Email address should start with First Letter of the First Name", TrFacesMessage.SEVERITY_ERROR);
        throw new TrValidatorException(fm);
    }

}

File: Chapter 8/Starter Applications/6.CreatingCustomCSValidator/ViewController/src/com/adffaces/chapter8/view/validators/EmailValidator.java

package com.adffaces.chapter8.view.validators;

import java.util.Collection;
import java.util.Collections;

import javax.faces.application.FacesMessage;
import javax.faces.component.UIComponent;
import javax.faces.context.FacesContext;
import javax.faces.validator.Validator;
import javax.faces.validator.ValidatorException;

import org.apache.myfaces.trinidad.validator.ClientValidator;
public class EmailValidator implements Validator {
    public EmailValidator() {
        super();
    }

    @Override
    public void validate(FacesContext facesContext, UIComponent uIComponent, Object object) throws ValidatorException {
        String firstName = uIComponent.getAttributes().get("firstName").toString().toUpperCase();
        if (!object.toString().toUpperCase().startsWith(firstName.charAt(0) + "")) {
            throw new ValidatorException(new FacesMessage(FacesMessage.SEVERITY_ERROR, "Validation Error",
                                                          "Email address should start with First Letter of the First Name"));
        }
    }
}
